File: src/logan/pickpocket/commands/ReloadCommand.java
package logan.pickpocket.commands;

import logan.config.MessageConfiguration;
import logan.pickpocket.main.PickpocketPlugin;
import logan.pickpocket.user.PickpocketUser;
import org.bukkit.entity.Player;

import java.util.List;

public class ReloadCommand implements PickpocketCommand {
    @Override
    public <T> void execute(Player player, List<PickpocketUser> profiles, T... args) {
        PickpocketPlugin.getPickpocketConfiguration().reload();
        PickpocketPlugin.getMessageConfiguration().reload();
        player.sendMessage(PickpocketPlugin.getMessageConfiguration().getMessage(MessageConfiguration.RELOAD_NOTIFICATION_KEY));
    }
}

File: src/main/java/logan/pickpocket/commands/ReloadCommand.kt
package logan.pickpocket.commands

import logan.api.command.BasicCommand
import logan.api.command.SenderTarget
import logan.pickpocket.config.MessageConfiguration
import logan.pickpocket.config.Config
import logan.pickpocket.main.PickpocketPlugin
import org.bukkit.command.CommandSender

class ReloadCommand : BasicCommand<CommandSender>(
    name = "reload",
    parentCommand = "pickpocket",
    target = SenderTarget.BOTH,
    permissionNode = "pickpocket.reload",
    usage = """
        Usage:
        /pickpocket reload (no args)
    """.trimIndent()
) {
    override fun run(sender: CommandSender, args: Array<out String>, data: Any?): Boolean {
        with (PickpocketPlugin.Companion) {
            Config.reload()
            MessageConfiguration.reload()
            profileConfiguration.reload()
            sender.sendMessage(MessageConfiguration.reloadNotificationMessage)
        }
        return true
    }
}